app/src/main/java/com/poupa/vinylmusicplayer/preferences/SharedPreferencesExporter.java

```diff
@@ -0,0 +1,97 @@
+package com.poupa.vinylmusicplayer.preferences;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.pm.PackageManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.ParcelFileDescriptor;
+
+import com.google.gson.Gson;
+import com.poupa.vinylmusicplayer.R;
+import com.poupa.vinylmusicplayer.util.OopsHandler;
+import com.poupa.vinylmusicplayer.util.SafeToast;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.preference.PreferenceManager;
+
+public class SharedPreferencesExporter extends AppCompatActivity {
+    private Context context;
+    private static final String FILENAME = "filename";
+    private ActivityResultLauncher<String> exportFilePicker;
+    private SharedPreferences sharedPreferences;
+
+    public static void start(Context context, String filename) {
+        Intent intent = new Intent(context, SharedPreferencesExporter.class);
+        intent.putExtra(FILENAME, filename);
+        context.startActivity(intent);
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        this.context = this.getApplicationContext();
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this.context);
+        Bundle b = this.getIntent().getExtras();
+        exportFilePicker = registerForActivityResult(new ActivityResultContracts.CreateDocument("text/plain"), result -> {
+            // Unless the selection has been cancelled, create the export file
+            if(result != null) {
+                try {
+                    writeToExportFile(result);
+                } catch (PackageManager.NameNotFoundException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            // Finishes the last activity to return to the settings activity.
+            this.finish();
+        });
+        exportFilePicker.launch(b.getString(FILENAME));
+    }
+
+    private void writeToExportFile(Uri location) throws PackageManager.NameNotFoundException {
+        Gson gson = new Gson();
+        HashMap<String, Object> prefsMap = new HashMap<>(sharedPreferences.getAll());
+        HashMap<String, Object> copyMap = new HashMap<>(prefsMap);
```

Comment:
Nitpicking here: You could have made only one copy of the `sharedPrefs.getAll`, instead of creating two here.